tests/accounts/test_csrf.py

```diff
@@ -0,0 +1,29 @@
+from django.conf import settings
+from django.test import Client
+from django.urls import reverse
+
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def _add_installed_apps(settings):
+    settings.LOGIN_URLS = [reverse("admin:login")]
+    settings.CSRF_FAILURE_VIEW = "maykin_common.accounts.csrf.csrf_failure"
+
+
+@pytest.mark.django_db
+def test_redirect_on_csrf_failure_if_logged_in(django_user_model):
+    user = django_user_model.objects.create_superuser(username="one")
+    client = Client(enforce_csrf_checks=True)
+    client.force_login(user)
+    response = client.post(reverse("admin:login"), {})
+
+    assert response.status_code == 302
+    assert response.url == settings.LOGIN_REDIRECT_URL  # type: ignore
+
+
+@pytest.mark.django_db
+def test_normal_csrf_failure_when_not_logged_in():
+    client = Client(enforce_csrf_checks=True)
+    response = client.post(reverse("admin:login"), {})
+    assert response.status_code == 403
```

Comment:
I'd add another assertion - `assertTemplateUsed` so that we're sure that we're looking at a CSRF failure view response and not some other error that happens to be a 403.

Reply:
ah yeah that makes sense then since it doesn't seem to [return a TemplateResponse](https://github.com/django/django/blob/6cb641ba75b1e6eace9a46e3cbade70e4af2ff66/django/views/csrf.py#L68).

This is the best we can do then :)